app/models/fe/element.rb

```diff
@@ -110,10 +110,10 @@ def previous_element(question_sheet, page = nil)
     end
 
     def hidden_by_conditional?(answer_sheet, page, prev_el)
-      (prev_el ||= previous_element(answer_sheet.question_sheet, page)) &&
-        prev_el.is_a?(Fe::Question) &&
-        prev_el.conditional == self &&
-        !prev_el.conditional_match(answer_sheet)
+      !!((prev_el ||= previous_element(answer_sheet.question_sheet, page)) &&
```

Comment:
I'm not a fan of the double bang, mostly because it can be hard to figure out exactly what it's doing without some mental flips. This would be more clear by creating some intermediate variables.